spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/common/HazelcastIntegrationUtils.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.hazelcast.common;
+
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+
+/**
+ * This is common utility class for SI-Hazelcast Support feature.
+ * 
+ * @author Eren Avsarogullari
+ * @since 1.0.0
+ *
+ */
+public abstract class HazelcastIntegrationUtils {
+	
+	/**
+	 * This constructor is added to prevent instantiation.
+	 * 
+	 */
+	private HazelcastIntegrationUtils() {
+
+    }
+
+	public static void shutdownAllHazelcastInstances() {
+		if (!Hazelcast.getAllHazelcastInstances().isEmpty()) {
+			for (HazelcastInstance hazelcastInstance : Hazelcast.getAllHazelcastInstances()) {
+				hazelcastInstance.shutdown();
```

Comment:
Why `shutdown` all, but not only those which are part of the current application context?
From other side I don't see the reason of this class at all.
Since we really have this `shutdown()` method we can use it from the `destroy-method` attribute on the `class="com.hazelcast.core.Hazelcast"` bean definition.
So, I'd leave it to end-users to destroy their `hazelcastInstance` on their own.